gammapy/maps/wcsnd.py

```diff
@@ -688,3 +692,68 @@ def cutout(self, position, width, mode="trim"):
         data = self.data[cutout_slices]
 
         return self._init_copy(geom=geom, data=data)
+
+    def sample_position_energy(self, npred_map=None, n_events=None, random_state=0):
+        """Sample position and energy of events.
+
+        Parameters
+        ----------
+        npred_map : `~gammapy.maps.Map`
+            Predicted number of counts map.
+        n_events : int
+            Number of events to sample.
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+
+        Returns
+        -------
+        coords : `~gammapy.maps.MapCoord` object.
+            Sequence of coordinates and energies of the sampled events.
+        """
+
+        self.random_state = get_random_state(random_state)
+        self.npred_map = npred_map
+        self.n_events = n_events
```

Comment:
Again no need to define a new attribute `self.n_events`, just use the local variable `n_events` later...